From ef353584005aa4afd748aef3efc97e1dec27ae97 Mon Sep 17 00:00:00 2001
From: thermatk <thermatk@thermatk.com>
Date: Fri, 22 Jun 2018 17:03:27 +0200
Subject: [PATCH 01/21] [CF][GMS] pass gn gen

most of the patch is made by `grep -rl '$google_play_services_package' ./ | xargs sed -i -e '/$google_play_services_package/d'`
---
 build/config/android/config.gni               |   6 -
 .../third_party/android_tools/BUILD.gn        | 150 ------------------
 chrome/android/BUILD.gn                       |  17 --
 chrome/test/android/BUILD.gn                  |   2 -
 chrome/test/android/cast_emulator/BUILD.gn    |   1 -
 components/background_task_scheduler/BUILD.gn |  13 --
 .../gcm_driver/instance_id/android/BUILD.gn   |   2 -
 .../signin/core/browser/android/BUILD.gn      |   3 -
 device/BUILD.gn                               |   3 -
 device/geolocation/BUILD.gn                   |   3 -
 services/BUILD.gn                             |   8 -
 services/shape_detection/BUILD.gn             |   4 -
 third_party/cacheinvalidation/BUILD.gn        |   3 -
 third_party/robolectric/BUILD.gn              |   3 -
 14 files changed, 218 deletions(-)

diff --git a/build/config/android/config.gni b/build/config/android/config.gni
index 75417a9c0c..241df13f84 100644
--- a/build/config/android/config.gni
+++ b/build/config/android/config.gni
@@ -93,12 +93,6 @@ if (is_android) {
     default_android_keystore_password = "chromium"
   }
 
-  # google_play_services_package contains the path where individual client
-  # targets (e.g. google_play_services_base_java) are located.
-  if (!defined(google_play_services_package)) {
-    google_play_services_package = "//third_party/android_tools"
-  }
-
   if (!defined(android_support_library_package)) {
     android_support_library_package = "//third_party/android_tools/support"
   }
diff --git a/build/secondary/third_party/android_tools/BUILD.gn b/build/secondary/third_party/android_tools/BUILD.gn
index 172aeb5d0b..38688d48b7 100644
--- a/build/secondary/third_party/android_tools/BUILD.gn
+++ b/build/secondary/third_party/android_tools/BUILD.gn
@@ -103,154 +103,4 @@ if (enable_java_templates) {
     android_deps_target_name = "com_android_support_support_v13_java"
     fallback_target = "$android_support_library_package:$target_name"
   }
-
-  # TODO(dgn): Use the POM files instead of hardcoding the dependencies.
-  gms_path = "$default_extras_android_sdk_root/extras/google/m2repository/com/google/android/gms"
-  gms_version = "11.2.0"
-
-  android_aar_prebuilt("google_play_services_basement_java") {
-    deps = [
-      ":android_support_v4_java",
-    ]
-    _lib_name = "play-services-basement"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    input_jars_paths = [ "$android_sdk/optional/org.apache.http.legacy.jar" ]
-  }
-
-  android_aar_prebuilt("google_play_services_tasks_java") {
-    deps = [
-      ":google_play_services_basement_java",
-    ]
-    _lib_name = "play-services-tasks"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_base_java") {
-    deps = [
-      ":android_support_v4_java",
-      ":google_play_services_basement_java",
-      ":google_play_services_tasks_java",
-    ]
-    _lib_name = "play-services-base"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_auth_base_java") {
-    deps = [
-      ":google_play_services_base_java",
-      ":google_play_services_basement_java",
-    ]
-    _lib_name = "play-services-auth-base"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_auth_java") {
-    deps = [
-      ":google_play_services_auth_base_java",
-      ":google_play_services_base_java",
-      ":google_play_services_basement_java",
-    ]
-    _lib_name = "play-services-auth"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_cast_java") {
-    deps = [
-      ":android_support_v7_mediarouter_java",
-      ":google_play_services_base_java",
-      ":google_play_services_basement_java",
-    ]
-    _lib_name = "play-services-cast"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_iid_java") {
-    deps = [
-      ":google_play_services_base_java",
-      ":google_play_services_basement_java",
-    ]
-    _lib_name = "play-services-iid"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_gcm_java") {
-    deps = [
-      ":google_play_services_base_java",
-      ":google_play_services_basement_java",
-      ":google_play_services_iid_java",
-    ]
-    _lib_name = "play-services-gcm"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_location_java") {
-    deps = [
-      ":google_play_services_base_java",
-      ":google_play_services_basement_java",
-    ]
-    _lib_name = "play-services-location"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_nearby_java") {
-    deps = [
-      ":google_play_services_base_java",
-      ":google_play_services_basement_java",
-    ]
-    _lib_name = "play-services-nearby"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_vision_java") {
-    deps = [
-      ":google_play_services_base_java",
-      ":google_play_services_basement_java",
-      ":google_play_services_vision_common_java",
-    ]
-    _lib_name = "play-services-vision"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
-
-  android_aar_prebuilt("google_play_services_vision_common_java") {
-    deps = [
-      ":google_play_services_base_java",
-      ":google_play_services_basement_java",
-    ]
-    _lib_name = "play-services-vision-common"
-    aar_path = "$gms_path/$_lib_name/$gms_version/$_lib_name-$gms_version.aar"
-    info_path = "//build/secondary/third_party/android_tools/$target_name.info"
-    proguard_configs =
-        [ "${target_gen_dir}/google_play_services_basement_java/proguard.txt" ]
-  }
 }
diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
index 89f8120669..f3a3399e3a 100644
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -195,14 +195,6 @@ android_library("chrome_java") {
     ":document_tab_model_info_proto_java",
     ":partner_location_descriptor_proto_java",
     ":thumbnail_cache_entry_proto_java",
-    "$google_play_services_package:google_play_services_auth_base_java",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
-    "$google_play_services_package:google_play_services_cast_java",
-    "$google_play_services_package:google_play_services_gcm_java",
-    "$google_play_services_package:google_play_services_iid_java",
-    "$google_play_services_package:google_play_services_nearby_java",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//chrome/android/third_party/compositor_animator:compositor_animator_java",
     "//chrome/android/webapk/libs/client:client_java",
@@ -436,9 +428,6 @@ junit_binary("chrome_junit_tests") {
     ":chrome_java_resources",
     ":class_register_java",
     ":partner_location_descriptor_proto_java",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
-    "$google_play_services_package:google_play_services_cast_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//base:base_junit_test_support",
@@ -512,12 +501,6 @@ android_library("chrome_test_java") {
 
   deps = [
     ":partner_location_descriptor_proto_java",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
-    "$google_play_services_package:google_play_services_cast_java",
-    "$google_play_services_package:google_play_services_gcm_java",
-    "$google_play_services_package:google_play_services_iid_java",
-    "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//chrome/android:app_hooks_java",
diff --git a/chrome/test/android/BUILD.gn b/chrome/test/android/BUILD.gn
index 7bb94ac81b..1c39d1c3fa 100644
--- a/chrome/test/android/BUILD.gn
+++ b/chrome/test/android/BUILD.gn
@@ -68,8 +68,6 @@ android_library("chrome_java_test_support") {
     "javatests/src/org/chromium/chrome/test/util/TranslateUtil.java",
   ]
   deps = [
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//base:base_java_test_support",
     "//chrome/android:chrome_java",
diff --git a/chrome/test/android/cast_emulator/BUILD.gn b/chrome/test/android/cast_emulator/BUILD.gn
index 0c36233a67..233f043f7a 100644
--- a/chrome/test/android/cast_emulator/BUILD.gn
+++ b/chrome/test/android/cast_emulator/BUILD.gn
@@ -20,7 +20,6 @@ android_library("cast_emulator_java") {
     "src/org/chromium/chrome/browser/media/router/DummyRoutePublisher.java",
   ]
   deps = [
-    "$google_play_services_package:google_play_services_cast_java",
     "//base:base_java",
     "//third_party/android_tools:android_support_v7_appcompat_java",
     "//third_party/android_tools:android_support_v7_mediarouter_java",
diff --git a/components/background_task_scheduler/BUILD.gn b/components/background_task_scheduler/BUILD.gn
index 9525aff817..8929a97692 100644
--- a/components/background_task_scheduler/BUILD.gn
+++ b/components/background_task_scheduler/BUILD.gn
@@ -42,10 +42,6 @@ if (is_android) {
     ]
 
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_gcm_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//base:base_java",
       "//third_party/android_tools:android_support_annotations_java",
     ]
@@ -61,10 +57,6 @@ if (is_android) {
 
     deps = [
       ":background_task_scheduler_java",
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_gcm_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//third_party/android_support_test_runner:runner_java",
@@ -85,11 +77,6 @@ if (is_android) {
 
     deps = [
       ":background_task_scheduler_java",
-      "$google_play_services_package:google_play_services_auth_base_java",
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_gcm_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//base:base_junit_test_support",
diff --git a/components/gcm_driver/instance_id/android/BUILD.gn b/components/gcm_driver/instance_id/android/BUILD.gn
index 2fdc248b31..a8cc895f80 100644
--- a/components/gcm_driver/instance_id/android/BUILD.gn
+++ b/components/gcm_driver/instance_id/android/BUILD.gn
@@ -20,7 +20,6 @@ generate_jni("test_support_jni_headers") {
 
 android_library("instance_id_driver_java") {
   deps = [
-    "$google_play_services_package:google_play_services_iid_java",
     "//base:base_java",
   ]
 
@@ -33,7 +32,6 @@ android_library("instance_id_driver_java") {
 android_library("instance_id_driver_test_support_java") {
   deps = [
     ":instance_id_driver_java",
-    "$google_play_services_package:google_play_services_iid_java",
     "//base:base_java",
   ]
 
diff --git a/components/signin/core/browser/android/BUILD.gn b/components/signin/core/browser/android/BUILD.gn
index ad4424314e..0e3fd647e9 100644
--- a/components/signin/core/browser/android/BUILD.gn
+++ b/components/signin/core/browser/android/BUILD.gn
@@ -13,9 +13,6 @@ generate_jni("jni_headers") {
 
 android_library("java") {
   deps = [
-    "$google_play_services_package:google_play_services_auth_base_java",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//base:base_java",
     "//net/android:net_java",
     "//third_party/android_tools:android_support_annotations_java",
diff --git a/device/BUILD.gn b/device/BUILD.gn
index 6982c5b52f..55f09b7b1e 100644
--- a/device/BUILD.gn
+++ b/device/BUILD.gn
@@ -344,9 +344,6 @@ if (is_android) {
       "geolocation/android/junit/src/org/chromium/device/geolocation/LocationProviderTest.java",
     ]
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_location_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//base:base_junit_test_support",
diff --git a/device/geolocation/BUILD.gn b/device/geolocation/BUILD.gn
index a69729c616..51ee4fcd03 100644
--- a/device/geolocation/BUILD.gn
+++ b/device/geolocation/BUILD.gn
@@ -127,9 +127,6 @@ if (is_android) {
     deps = [
       ":geolocation",
       ":geolocation_jni_headers",
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_location_java",
       "//base:base_java",
     ]
   }
diff --git a/services/BUILD.gn b/services/BUILD.gn
index 1da26c6e46..9ed2bf30e0 100644
--- a/services/BUILD.gn
+++ b/services/BUILD.gn
@@ -138,10 +138,6 @@ if (is_android) {
       "shape_detection/android/junit/src/org/chromium/shape_detection/BitmapUtilsTest.java",
     ]
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_vision_common_java",
-      "$google_play_services_package:google_play_services_vision_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//base:base_junit_test_support",
@@ -167,10 +163,6 @@ if (is_android) {
       "shape_detection/android/javatests/src/org/chromium/shape_detection/TextDetectionImplTest.java",
     ]
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_vision_common_java",
-      "$google_play_services_package:google_play_services_vision_java",
       "//base:base_java",
       "//base:base_java_test_support",
       "//mojo/public/java:bindings_java",
diff --git a/services/shape_detection/BUILD.gn b/services/shape_detection/BUILD.gn
index 6debb79625..e4445afcfc 100644
--- a/services/shape_detection/BUILD.gn
+++ b/services/shape_detection/BUILD.gn
@@ -89,10 +89,6 @@ if (is_android) {
     ]
 
     deps = [
-      "$google_play_services_package:google_play_services_base_java",
-      "$google_play_services_package:google_play_services_basement_java",
-      "$google_play_services_package:google_play_services_vision_common_java",
-      "$google_play_services_package:google_play_services_vision_java",
       "//base:base_java",
       "//mojo/android:system_java",
       "//mojo/public/java:bindings_java",
diff --git a/third_party/cacheinvalidation/BUILD.gn b/third_party/cacheinvalidation/BUILD.gn
index fd4d77bccf..2386c55cd4 100644
--- a/third_party/cacheinvalidation/BUILD.gn
+++ b/third_party/cacheinvalidation/BUILD.gn
@@ -137,9 +137,6 @@ if (is_android) {
     emma_never_instrument = true
     deps = [
       ":cacheinvalidation_proto_java",
-      "$google_play_services_package:google_play_services_gcm_java",
-      "$google_play_services_package:google_play_services_iid_java",
-      "$google_play_services_package:google_play_services_tasks_java",
       "//third_party/android_protobuf:protobuf_nano_javalib",
       "//third_party/android_tools:android_gcm_java",
     ]
diff --git a/third_party/robolectric/BUILD.gn b/third_party/robolectric/BUILD.gn
index 935065fecf..dee8cd7572 100644
--- a/third_party/robolectric/BUILD.gn
+++ b/third_party/robolectric/BUILD.gn
@@ -746,9 +746,6 @@ java_library("shadows_play_services_java") {
     ":robolectric_shadowapi_java",
     ":robolectric_utils_java",
     ":shadows_core_java",
-    "$google_play_services_package:google_play_services_auth_base_java",
-    "$google_play_services_package:google_play_services_base_java",
-    "$google_play_services_package:google_play_services_basement_java",
     "//third_party/android_tools:android_support_v4_java",
     "//third_party/gson:gson_java",
     "//third_party/guava:guava_java",
-- 
2.17.1

